### PR TITLE
Armor flag support on key export command

### LINKED ISSUE #3279: Support for 'key export --armor' (Exporting Keys in ASCII Armored File)
## Version of Singularity:

```
3.1.1-902.ge094ee448
```

### Expected behavior

```bash
$ singularity key export --secret --armor ./file
[...]
Enter # of signing key to use : 0
Enter key passphrase : 
Private key with fingerprint 21490E395B8DF83CB8C2669AF426CAF995116EAE correctly exported to file: ./file

$ cat file
-----BEGIN PGP PUBLIC KEY BLOCK-----

mQGNBFynqCEBDACldwWSm8oF7wE8q0d+GnCYIyOFiohi+VI5Q4ianstwmuXl8WLD
BaM6elodQLaj7rjiMVsWY6AAwE7YbapeL8K66Nms384/ArTt6Qgn1p8na63j+yjX
[...]
-----END PGP PUBLIC KEY BLOCK-----
```

### Actual behavior

Does not exist yet.

### Steps to reproduce behavior

```bash
$ singularity key export --secret --armor ./file
```

<br>

### PR DESCRIPTION
**Pull request by jchavezlavalle**
**Description of the Pull Request (PR):**

Support for armor flag on key export command.

**This fixes or addresses the following GitHub issues:**

- Fixes #3279 

**Before submitting a PR, make sure you have done the following:**

- Read the [Guidelines for Contributing](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md), and this PR conforms to the stated requirements.
- Added changes to the [CHANGELOG](https://github.com/sylabs/singularity/blob/master/CHANGELOG.md) if necessary according to the [Contribution Guidelines](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md)
- Added tests to validate this PR and tested this PR locally with a `make testall`
- Based this PR against the appropriate branch according to the [Contribution Guidelines](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md)
- Added myself as a contributor to the [Contributors File](https://github.com/sylabs/singularity/blob/master/CONTRIBUTORS.md)

Attn: @singularity-maintainers

### CODE REVIEW

File: cmd/internal/cli/key_export.go [hunk 1]
```diff
@@ -14,14 +14,14 @@ import (
...
-	secretExport bool
-)
+var secretExport bool
+var armor bool
 
 func init() {
 	KeyExportCmd.Flags().SetInterspersed(false)
 
 	KeyExportCmd.Flags().BoolVarP(&secretExport, "secret", "s", false, "export a secret key")
+	KeyExportCmd.Flags().BoolVarP(&armor, "armor", "a", false, "key armored format")
```
Comment (mem) [inline cmd/internal/cli/key_export.go]: maybe `use ascii armored format`?
```suggestion
	KeyExportCmd.Flags().BoolVarP(&armor, "armor", "a", false, "use ascii armored format")
```

**Review by WestleyK — CHANGES_REQUESTED** [review]
Looks very good! Although we need a way to import the ascii keys. It should detect what type of key it is (binary or ascii) automatically.

**Comment by WestleyK** [comment]
Let me know if you have any questions or need help :+1:

**Review by mem — APPROVED** [review]

**Comment by mem** [comment]
> Looks very good! Although we need a way to import the ascii keys. It should detect what type of key it is (binary or ascii) automatically.

Can we push that one to a different PR?

**Comment by WestleyK** [comment]
Yes. But there wont be a way to import a ascii key. (We can still import binary keys) :+1: save it for another PR!

**Review by WestleyK — APPROVED** [review]
Looks good to me! :+1:

**Pull request merged**